authenticate.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+from getpass import getpass
+import logging
+from pathlib import Path
+
+import credentials
+from bwproject import BWUser
+
+import argparse
+
+
+def authenticate(username, password, credentials_path=None):
+    """""
+    Authenticate the given username and password pair, storing the access token in the credentials file.
+
+    :param username: Brandwatch account usernames
+    :param password: Brandwatch account password
+    :param credentials_path: Path to where credentials should be stored
+    :return: An authenticated BWUser object
+    """
+    return BWUser(username=username, password=password, token_path=credentials_path)
+
+
+def main():
+    logger = logging.getLogger("bwapi")
+    logger.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter("%(levelname)s: %(message)s", "%H:%M:%S")
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    parser = argparse.ArgumentParser(description="Logging to Brandwatch and retrieve and access token.")
+
+    parser.add_argument("--store", "-s", type=Path, metavar="PATH", default=credentials.DEFAULT_CREDENTIALS_PATH,
+                        help="Path to where access tokens are stored. (Default: {})".format(
+                            credentials.DEFAULT_CREDENTIALS_PATH))
```

Comment:
You can use [ArgumentDefaultsHelpFormatter](https://docs.python.org/3/library/argparse.html#formatter-class) to show the defaults for arguments.

Reply:
Ah nice, thanks :-)